roundabout-core/src/main/java/at/fhv/itm3/s2/roundabout/entity/StreetSection.java

```diff
@@ -47,21 +97,56 @@ public IStreetConnector getPreviousStreetConnector() {
 
     @Override
     public Map<ICar, Double> getCarPositions() {
-        return carPositions;
+        return Collections.unmodifiableMap(carPositions);
     }
 
+    @Override
     public void updateAllCarsPositions() {
-
+        final double currentTime = getCurrentTime();
+
+        // Updating positions for all cars.
+        ICar previousCar = null;
+        for (ICar currentCar : carQueue) {
+            final double carLastUpdateTime = currentCar.getLastUpdateTime();
+            final double carSpeed = currentCar.getDriverBehaviour().getSpeed();
+            final double carPosition = getCarPositions().getOrDefault(currentCar, INITIAL_CAR_POSITION);
+
+            // Calculate distance to next car / end of street section based on distributed driver behaviour values.
+            final double distanceToNextCar = 0; //TODO finish.
+            currentCar.getDriverBehaviour().getMinDistanceToNextCar();
+            currentCar.getDriverBehaviour().getMaxDistanceToNextCar();
```

Comment:
why? no assignment to a variable or smth...

Reply:
not finished yet. lines 115-117 will be refactored when i will add a random distribution generator to model.